### DISCUSSION TITLE
麻烦请教下Windows环境VS2017下的链接问题

### DISCUSSION THREAD

**Discussion by FlyingPork** [k2-fsa/sherpa-onnx#293] [Q&A]
cmake时指定 -G "Visual Studio 15 2017" 与 -DCMAKE_GENERATOR_TOOLSET=v141，结果在链接时

c-api-examples\decode-file-c-api.vcxproj 、

sherpa-onnx\csrc\sherpa-onnx.vcxproj、

sherpa-onnx\csrc\sherpa-onnx-microphone.vcxproj

等一系列项目均出现了如下问题：

onnxruntime_mlas.lib(pooling.obj) : error LNK2001: 无法解析的外部符号 __GSHandlerCheck_EH4 [G:\sherpa-onnx\build\c-api-examples\decode-file-c-api.vcxproj]（类似的报错还有出现在其它lib上）

麻烦问下这个是否是onnxruntime-win-x64-static-1.15.1.tar.bz2版本太高导致的，换成onnxruntime-win-x64-static-1.14.0.tar.bz2是否可以正常在VS2017下编译与链接？或者安装更新的VS版本是更好的选择？

**Comment by csukuangfj** [comment]
建议使用 vs 2022 试试。

如果 vs 2022 可以，那么你可以考虑使用 vs 2017 自己去编译一个 onnxruntime 的库.

(或者，你试试动态库?)

**Comment by csukuangfj** [comment]
我试过 vs2022, 静态链接完全没问题.

onnxruntime 的静态库，是由
https://github.com/csukuangfj/onnxruntime-libs/actions/runs/5542318075/job/15010659820
编译的。你可以 fork 一下，换成 vs2017 去编译。

**Comment by FlyingPork** [comment]
好的，我试试

**Reply by csukuangfj** [reply]
解决了吗？

**Reply by FlyingPork** [reply]
嗯，已经解决，VS2022可以正常编译与链接。现在正在尝试重新编译onnxruntime 的库--试下sherpa-onnx Windows下的Debug版本进行调试。

多谢提供的信息
